docs/IWYUStyle.md

```diff
@@ -0,0 +1,242 @@
+# IWYU Coding Style #
```

Comment:
Should this also mention how we work with Git? 
Like, what to put in commit messages, splitting functionality, rebase instead of merge.. etc?